src/main/java/org/spongepowered/api/plugin/Plugin.java

```diff
@@ -24,19 +24,21 @@
  */
 package org.spongepowered.api.plugin;
 
-import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import java.util.regex.Pattern;
 
 /**
  * An annotation used to describe and mark a Sponge plugin.
  */
-@Target(TYPE)
-@Retention(RUNTIME)
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
 public @interface Plugin {
 
+    Pattern ID_PATTERN = Pattern.compile("[a-z0-9-_]+");
```

Comment:
I doubt Forge does any limitation here, but while we can't allow more, we can always allow less, it would only limit the choices of the ID, but don't have any functional impact